lalrpop/src/normalize/token_check/mod.rs

```diff
@@ -56,13 +87,16 @@ struct Validator<'grammar> {
     grammar: &'grammar Grammar,
     all_literals: Map<TerminalLiteral, Span>,
     conversions: Option<Set<TerminalString>>,
+    match_mappings: Option<Map<TerminalString, TerminalString>>,
+    match_catch_all: Option<bool>,
```

Comment:
instead of having two option fields that are either both some or both none, maybe this:

```rust
/// Summarizes the user's `match` declaration, if any:
///
/// Example:
///     match {
///         "foo" => bar,
///         "baz" => "qux",
///         _
///     };
/// would result in a "rev_mapping" of `bar` to `"foo"` and `"qux"` to `"baz"`.
///
/// The `catch_all` flag would be true, since the `_` is present. 
struct MatchData {
    rev_mapping: Map<TerminalString, TerminalString>,
    catch_all: bool,
}
```